extensions/azurecore/src/account-provider/interfaces.ts

```diff
@@ -221,7 +221,7 @@ export interface AzureAccountSecurityToken {
  */
 export type AzureAccountSecurityTokenCollection = { [tenantId: string]: AzureAccountSecurityToken };
 
-export interface Deferred<T> {
+export interface Deferred<T, E extends Error> {
```

Comment:
`E extends Error = Error`? That way they don't have to define it if they just want to use the base Error type